### PR TITLE
less constructors

### PR DESCRIPTION
**Pull request by FinnStutzenstein**

### CODE REVIEW

File: client/src/app/shared/models/motions/workflow-state.ts [hunk 1]
```diff
@@ -27,57 +27,12 @@ export class WorkflowState implements Deserializable {
...
-        this.allow_create_poll = allow_create_poll;
-        this.allow_submitter_edit = allow_submitter_edit;
-        this.versioning = versioning;
-        this.leave_old_version_active = leave_old_version_active;
-        this.dont_set_identifier = dont_set_identifier;
-        this.show_state_extension_field = show_state_extension_field;
-        this.show_recommendation_extension_field = show_recommendation_extension_field;
-        this.next_states_id = next_states_id;
-        this.workflow_id = workflow_id;
+    public constructor(input?: any) {
```
Comment (tsiegleauq) [inline client/src/app/shared/models/motions/workflow-state.ts]: deserialize needs adjustment here

File: client/src/app/shared/models/motions/workflow.ts [hunk 2]
```diff
@@ -12,13 +12,12 @@ export class Workflow extends BaseModel {
...
-    public constructor(id?: number, name?: string, states?: WorkflowState[], first_state?: number) {
+    public constructor(input?: any) {
         super();
         this._collectionString = 'motions/workflow';
-        this.id = id;
-        this.name = name;
-        this.states = states;
-        this.first_state = first_state;
+        if (input) {
+            this.deserialize(input);
```
Comment (tsiegleauq) [inline client/src/app/shared/models/motions/workflow.ts]: deserialize needs adjustment here

File: client/src/app/shared/models/mediafiles/mediafile.ts [hunk 3]
```diff
@@ -16,31 +16,17 @@ export class Mediafile extends BaseModel {
...
-        this.id = id;
-        this.title = title;
-        this.mediafile = mediafile;
-        this.media_url_prefix = media_url_prefix;
-        this.uploader_id = uploader_id;
-        this.filesize = filesize;
-        this.hidden = hidden;
-        this.timestamp = timestamp;
+        if (input) {
+            this.deserialize(input);
```
Comment (tsiegleauq) [inline client/src/app/shared/models/mediafiles/mediafile.ts]: deserialize needs adjustment here

File: client/src/app/shared/models/assignments/assignment.ts [hunk 4]
```diff
@@ -21,30 +21,15 @@ export class Assignment extends BaseModel {
...
-        this.assignment_related_users = assignment_related_users || []; // TODO Array
-        this.poll_description_default = poll_description_default;
-        this.polls = polls || Array(); // TODO Array
-        this.agenda_item_id = agenda_item_id;
-        this.tags_id = tags_id;
+        this.assignment_related_users = []; // TODO Array
+        this.polls = Array(); // TODO Array
+
+        if (input) {
+            this.deserialize(input);
```
Comment (tsiegleauq) [inline client/src/app/shared/models/assignments/assignment.ts]: deserialize needs adjustment here

File: client/src/app/shared/models/agenda/item.ts [hunk 5]
```diff
@@ -28,38 +28,12 @@ export class Item extends BaseModel {
...
-        this.type = type;
-        this.is_hidden = is_hidden;
-        this.duration = duration;
-        this.speakers = speakers;
-        this.speaker_list_closed = speaker_list_closed;
-        this.content_object = content_object;
-        this.weight = weight;
-        this.parent_id = parent_id;
+        if (input) {
+            this.deserialize(input);
```
Comment (tsiegleauq) [inline client/src/app/shared/models/agenda/item.ts]: deserialize needs adjustmnet here

File: client/src/app/shared/models/motions/workflow-state.ts (hunk 1 above)
Comment (FinnStutzenstein) [reply client/src/app/shared/models/motions/workflow-state.ts]: Done.

File: client/src/app/shared/models/motions/workflow.ts (hunk 2 above)
Comment (FinnStutzenstein) [reply client/src/app/shared/models/motions/workflow.ts]: Done.

File: client/src/app/shared/models/mediafiles/mediafile.ts (hunk 3 above)
Comment (FinnStutzenstein) [reply client/src/app/shared/models/mediafiles/mediafile.ts]: Done.

File: client/src/app/shared/models/assignments/assignment.ts (hunk 4 above)
Comment (FinnStutzenstein) [reply client/src/app/shared/models/assignments/assignment.ts]: Done.

File: client/src/app/shared/models/agenda/item.ts (hunk 5 above)
Comment (FinnStutzenstein) [reply client/src/app/shared/models/agenda/item.ts]: Done.

File: client/src/app/shared/models/motions/motion.ts [hunk 6]
```diff
@@ -44,49 +44,20 @@ export class Motion extends BaseModel {
...
+        this.origin = '';
+        this.submitters = [];
+        this.supporters_id = [];
+        this.state_required_permission_to_see = '';
+        this.log_messages = [];
 
+        if (input) {
+            this.deserialize(input);
+        }
         this.initDataStoreValues();
```
Comment (tsiegleauq) [inline client/src/app/shared/models/motions/motion.ts]: this function usually fils with
`Cannot read property 'isStateContained' of undefined`

Accessing variables is not safe anymore.

**Comment by FinnStutzenstein** [comment]
@tsiegleauq Please see last commits

**Pull request merged**